### PR TITLE
[FEC-49] - add type definition model

### PR DESCRIPTION
**Pull request by antoniolodias**
#### Summary

This PR creates `type definition` model needed for this [migration PR](https://github.com/commercetools/merchant-center-frontend/pull/17434) which is based on [batch 12](https://commercetools.atlassian.net/browse/FEC-49) of the [chapter epic](https://commercetools.atlassian.net/browse/FEC-26) where we are migrating tests to use public test-data package

### CODE REVIEW

File: models/product-type/src/type-definition/types.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// import { TypeDefinition } from '@commercetools/platform-sdk'; shouldn't this come from here?
+import type { TBuilder } from '@commercetools-test-data/core';
+
+export interface TypeDefinition {
+  id: string;
+  key: string;
+  name: null;
+  fieldDefinitions: null;
+}
```
Comment (jaikumar-tj) [inline models/product-type/src/type-definition/types.ts:9]: Looks like there are more fields 
Screenshot 2024-09-04 at 05 29 54

File: models/product-type/src/type-definition/transformers.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    buildFields: ['fieldDefinitions', 'name'],
+  }),
+  rest: Transformer<TTypeDefinition, TTypeDefinition>('rest', {
+    buildFields: ['fieldDefinitions', 'name'],
+  }),
+  graphql: Transformer<TTypeDefinition, TTypeDefinitionGraphql>('graphql', {
+    buildFields: ['fieldDefinitions'],
+    replaceFields: ({ fields }) => {
+      const nameAllLocales = LocalizedString.toLocalizedField(fields.name);
+
```
Comment (jaikumar-tj) [inline models/product-type/src/type-definition/transformers.ts:16]: Should we transform the field `name` using `LocalizedString.resolveGraphqlDefaultLocaleValue` ?

[For example](https://github.com/commercetools/test-data/blob/a3ccfca0a4cc94eb6bcae6ccba050a9999c8f341/models/attribute-group/src/transformers.ts#L19)

File: models/product-type/src/type-definition/builder.spec.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    const required = true;
+
+    const name = LocalizedString.random().en(nameEn);
+    const fieldDefinitions = FieldDefinition.random().required(required);
+
+    const typeDefinitionMock = TypeDefinition.random()
+      .id(id)
+      .key(key)
+      .name(name)
+      // TODO: fix TS error
```
Comment (CarlosCortizasCT) [inline models/product-type/src/type-definition/builder.spec.ts:65]: Could you share what's the error?

File: models/product-type/src/type-definition/types.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+// import { TypeDefinition } from '@commercetools/platform-sdk'; shouldn't this come from here?
```
Comment (CarlosCortizasCT) [inline models/product-type/src/type-definition/types.ts:1]: Does this type actually exists in the REST API?

File: models/product-type/src/type-definition/types.ts (hunk 1 above)
Comment (CarlosCortizasCT) [reply models/product-type/src/type-definition/types.ts]: Updated here: 63053ce8aeda6ddae51dcf9aea3bd837d03ce565

File: models/product-type/src/type-definition/transformers.ts (hunk 2 above)
Comment (CarlosCortizasCT) [reply models/product-type/src/type-definition/transformers.ts:32]: Updated here: 63053ce8aeda6ddae51dcf9aea3bd837d03ce565

File: models/product-type/src/type-definition/builder.spec.ts (hunk 3 above)
Comment (CarlosCortizasCT) [reply models/product-type/src/type-definition/builder.spec.ts]: Updated here: 63053ce8aeda6ddae51dcf9aea3bd837d03ce565

File: models/product-type/src/type-definition/types.ts (hunk 4 above)
Comment (CarlosCortizasCT) [reply models/product-type/src/type-definition/types.ts]: It seems it does not.

**Review by CarlosCortizasCT — APPROVED** [review]
Looks good to me 👍 

Thanks Antonio 🙇

**Review by jaikumar-tj — APPROVED** [review]

**Pull request merged**